### DISCUSSION TITLE
getAvailablePurchases returning null in Amazon (i have to clean the store cache to get it working again)

### DISCUSSION THREAD

**Discussion by pedrolmsin** [hyochan/react-native-iap#2512] [Q&A]
Hi everyone, my app uses react-native-iap in the 3 stores, in the play store and apple app store everything is perfectly normal and functional, in amazon when I upload my app to live test and test the purchases everything is fine, but after a while the getAvailablePurchases and getPurchaseHistory functions start returning empty, they only start returning the user's purchases again when i clean the amazon app store cache, which leads me to believe that the function is taking data from the amazon app store cache which for some reason stores an array of empty objects, anyone with this problem?

Thanks.

**Comment by hyochan** [comment]
Thanks for raising this!

At the moment, Amazon IAP / Fire OS is not officially supported in react-native-iap due to the lack of a proper testing environment on my side.

I’ve shared more context and the current stance here:
https://github.com/hyochan/expo-iap/discussions/239

If anyone from the community has hands-on experience with Amazon IAP and would like to help (testing, documentation, or PRs), contributions are very welcome.

I’m going to close this discussion for now and will revisit Amazon IAP support if proper resources or official support become available. Thanks for your understanding!
